sycl/doc/UsersManual.md

```diff
@@ -436,6 +437,13 @@ and not recommended to use in production environment.
 
     NOTE: This flag is only supported for spir64_gen AOT targets.
 
+**`-ftarget-register-alloc-mode=<arg>`**
+
+    Specify a register allocation mode for specific hardware for use by supported
+    target backends. The format of the argument is "Device0:Mode0[,Device1:Mode1...]".
+    Currently the only supported Device is "pvc". The supported modes are
+    "default","small","large", and "auto".
```

Comment:
What is the difference between "default" and "auto". The description defines them as:
- auto -> request the backend to use heuristics to pick a register alloc mode
- default -> provide no specification to the backend on what register alloc mode to use.

I don't understand the difference between them from the user POV. In what cases users should set "auto" instead of "default"?

Reply:
@bader 

Thanks for the review.

`auto` means "analyze my code and try to pick the best one (large or small)"
`default` means "i don't care what you do, do whatever you do when the user doesn't care". 

Setting `auto` may increase performance if the heuristic used in the backend picks a mode that improves performance.
Setting `default` is the same as specifying nothing, you are saying you don't care what the backend does. It is the same as not passing the argument at all (we may change the default value of this argument to be `pvc:auto`, so in that case, if the heuristics are bad, `default` gives you a chance to disable the heuristics.

Reply:
To be honest, I don't see why default shouldn't be "auto". If user don't care, using "auto" seems to be the most reasonable choice. 
I would rather remove "default" value and say in the description that if option is not set the value is "auto".

Reply:
Well, that assumes auto doesn't cause regressions, which we haven't yet confirmed. I'll send you an message offline another reason too.

Reply:
"default" means "I don't care", right? So, regressions should not block us. 
If we care about regression, we can set the value to "small" if option is not specified.
Anyway, I don't see use cases for explicit "default" value.

Reply:
Discussed this offline, thanks for the discussion